src/runtime/update-component.ts

```diff
@@ -118,7 +118,11 @@ const dispatchHooks = (hostRef: d.HostRef, isInitialLoad: boolean): Promise<void
  * @returns either a `Promise` or the return value of the provided function
  */
 const enqueue = (maybePromise: Promise<void> | undefined, fn: () => Promise<void>): Promise<void> | undefined =>
-  maybePromise instanceof Promise ? maybePromise.then(fn) : fn();
+  isPromisey(maybePromise) ? maybePromise.then(fn) : fn();
+
+const isPromisey = (maybePromise: Promise<void> | unknown): maybePromise is Promise<void> =>
```

Comment:
There's a lot of good detail about this change in the pr summary - could we capture the cliff notes of it for `isPromisey` to describe why we're doing this structural type checking?

Reply:
ah made that change locally but forgot to push it! haha